Data/models/Bookmark.swift

```diff
@@ -280,29 +280,31 @@ public final class Bookmark: NSManagedObject, WebsitePresentable, Syncable, CRUD
         
         let context = DataController.newBackgroundContext()
         // Note: sync order is also used for ordering favorites and non synchronized bookmarks.
-        updateSyncOrderOfMovedBookmark(frc: frc, sourceBookmark: src,
-                                       destinationBookmark: dest,
-                                       sourceIndexPath: sourceIndexPath,
-                                       destinationIndexPath: destinationIndexPath,
-                                       context: context)
+        let srcUpdated = updateSyncOrderOfMovedBookmark(frc: frc, sourceBookmark: src,
+                                                        destinationBookmark: dest,
+                                                        sourceIndexPath: sourceIndexPath,
+                                                        destinationIndexPath: destinationIndexPath,
+                                                        context: context)
         
         setOrderForAllBookmarksOnGivenLevel(parent: src.parentFolder, forFavorites: src.isFavorite, context: context)
         
         DataController.save(context: context)
-        // Updating sync orders is not supported on server side yet, see issue #772 for more details.
-        if !src.isFavorite { Sync.shared.sendSyncRecords(action: .update, records: [src]) }
```

Comment:
The bug was caused by sending to sync record a bookmark which syncOrder hasn't been updated yet.
`src` is a Bookmark on main thread, taken from `frc`, and we were updating its order on a background thread.

1.9 with improved Core Data should help with such issues